modules/imgproc/src/smooth.simd.hpp

```diff
@@ -1233,7 +1233,15 @@ void hlineSmoothONa_yzy_a<uint16_t, ufixedpoint32>(const uint16_t* src, int cn,
     for (; i <= lencn - VECSZ * 2; i += VECSZ * 2, src += VECSZ * 2, dst += VECSZ * 2)
     {
         v_uint32 v_res0, v_res1;
-        v_mul_expand(vx_load(src + pre_shift * cn), vx_setall_u16((uint16_t) *((uint32_t*)(m + pre_shift))), v_res0, v_res1);
+        if (*(m + pre_shift) == ufixedpoint32::fromRaw(1 << 16))
```

Comment:
Lets do not slowdown this kernel.
This kernel assumes that kernel values are less than 1.0 (lets keep its behavior). GaussianBlur kernel has property that sum(kernel values) = 1. So all useful kernels passes this assumption.

Please add precheck for such special kernels and do not run SIMD (any) code for these trivial cases.

Reply:
I agree with you, it would be better to have an overall approach for uint types, for floats we can determine surely the equality. So for the unit type of matrixes and kernel should check the equality with 1 (MAX of uint type), if so then can skip any further processing just memcpy the source to destination. This approach make unnecessary the wider kernel types. Yes, it is not a small thing to consider. It is just a bug fix at this moment.